src/gurt/dlog.c

```diff
@@ -638,14 +638,13 @@ d_log_open(char *tag, int maxfac_hint, int default_mask, int stderr_mask,
 	/* cache value of isatty() to avoid extra system calls */
 	mst.stdout_isatty = isatty(fileno(stdout));
 	mst.stderr_isatty = isatty(fileno(stderr));
-	d_log_xst.tag = newtag;
+	d_log_xst.tag = tag;
```

Comment:
This is likely causing the crash - tag might not be available once this function exits, previously a copy was made in a 'newtag' which was allocated and kept around. 

Also on line 663 you can no longer do free(d_log_xst.tag) with the way you made a change here

Reply:
Oops, I should have worked too late/long... Don't know what I really wanted to optimize/simplify there! Thanks Alex.